src/pyenphase/envoy.py

```diff
@@ -310,9 +312,14 @@ async def open_dry_contact(self, id: str) -> dict[str, Any]:
                 "This feature is not available on this Envoy."
             )
 
-        return await self._json_request(
+        result = await self._json_request(
             URL_DRY_CONTACT_STATUS, {"dry_contacts": {"id": id, "status": "open"}}
         )
+        # The Envoy takes a few seconds before it will reflect the new state of the relay
+        # so we preemptively update it
+        if self.data:
+            self.data.dry_contact_status[id].status = DryContactStatus.OPEN
```

Comment:
```suggestion
        if data := self.data:
            data.dry_contact_status[id].status = DryContactStatus.OPEN
```

Reply:
Would save one lookup but realistically `self.data` should never be none here so you could do
```
if TYPE_CHECKING:
  assert self.data is not None
```